pkg/scheduler/objects/queue.go

```diff
@@ -1571,3 +1593,84 @@ func (sq *Queue) recalculatePriority() int32 {
 	sq.currentPriority = curr
 	return priorityValueByPolicy(sq.priorityPolicy, sq.priorityOffset, curr)
 }
+
+func (sq *Queue) UpdateApplicationPreemptionPriority(applicationID string, priority int32) {
+	if sq == nil || !sq.isLeaf {
```

Comment:
there is a locked method to check `sq.isLeaf` should be using that one

Reply:
Done